sklearn/impute.py

```diff
@@ -537,6 +558,38 @@ def _get_missing_features_info(self, X):
 
         return imputer_mask, features_with_missing
 
+    def _drop_constant_columns(self, X):
+        """
+        In case we have a column with totally missing values, drop it,
+        because constant column is not informative.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            The input data.
+
+        Returns
+        -------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            The input data without a column with totally missing values.
+        """
+        is_sparse = sparse.issparse(X)
+
+        if is_sparse:
+            sparse_format = X.format
+            X = X.toarray()
+
+        if not self.fully_missing_indexes_:
+            ix = _get_mask(X, self.missing_values)
```

Comment:
Is there a way we can simply reuse an already-calculated mask? This seems to be unnecessary duplication of work. Can we just look for statistics_ with value NaN?

Reply:
We can't just look for `statistics_` since for the `MissingIndicator` we want to drop the columns with no missing values (which is it's default behavior). This info can't be extracted from `statistics_`.

We could swith to the `features='all'` behavior but I think it's better to only keep the relevant features.

To avoid recomputing the mask, we could make `MissingIndicator` to accept a precomputed mask. What do you think about that ?